### ISSUE TITLE
Dependency trees not hashed correctly

### ISSUE THREAD

**Issue by vrugtehagel** [vrugtehagel/eleventy-asset-hash#1]
When there is a chain of files requesting each other, for example a JavaScript file A requesting a file B that requests C, then the plugin currently naively adds hashes to each URL, but does not resolve the dependency tree. This is a problem, because if I make a change in file C, then its hash changes, but B's hash does not. That means A will request the same version for B that it did before, which includes the old hash for C.

In a nutshell, the plugin needs to hash files _after_ having replaced all the URL they reference have been processed. The hashes need to be hashes of the files that already include URLs with hash parameters.

**Comment by vrugtehagel** [comment]
This issue has been partially resolved. The issue as outlined has been fixed, but circular dependencies are currenntly not handled correctly. They are hashed twice, which works for when the dependency loop is of size 2, but not beyond that. Instead, dependency loops should be identified and hashed as a whole; all files in a dependency loop should get the same hash, since all depend on one another. However, the loops need to be distinguished; that is, two separate dependency loops should _not_ get the same hash, just the files within one loop do.
